@here/harp-mapview/lib/text/TextElement.ts

```diff
@@ -609,4 +612,21 @@ export class TextElement {
             }
         }
     }
+
+    /**
+     * Something like a unique hash for each text element, the text element position, and uid are
+     * ignored
+     */
+    getIdentityKey(): string {
```

Comment:
As far as I can see the unique id is mainly based on the text. What if the label has no text (I guess this can happen for POIs) or even worse if the text is the same (this will happen for sure for POIs like gas stations).

Reply:
I can't identify labels between zoom levels because their position changes at each level.
This function generates a key by which I can place similar labels render states in one array, and then if needed we can find and copy a state for a new label at another zoom level.
I'll try to rename this method